Clarifying dependents in taxes with the new guardian information (#1041)
ihmeuw/vivarium_research: docs/source/concept_models/vivarium_census_synthdata/concept_model.rst

improve tracking for names, and dependent status on tax forms. 

This person will be listed as ["Guardian"]. By design, most will be 

> parents, but some may be a grandparent or other relative. This will 

Reviewer: IMO, describing the state table shouldn't be necessary if we clearly outline how this works. While I'm pretty sure in this case it will be implemented the way you describe here, it could also be implemented the opposite way (each simulant keeps track of who they are the guardian **of**) and that would also work.

Reviewer: I think the main thing this section needs is to consistently use the term "guardian" instead of "parent/guardian" -- because "parent" is already a concept in the simulation and we do not want them to be confused with each other.

Also, I would specify here at the top the restriction that each simulant will have a maximum of 2 guardians.

In the rules below, to make it more implementation-agnostic, we can be a bit more explicit, something like saying "assign the reference person to be this simulant's guardian" instead of "assign reference person." Or, we can add some introductory text that explains to interpret the rules this way.

A nit I didn't notice until now: "child"/"parent" are also used in the assignment rules below. This is particularly confusing on [line 421](https://github.com/ihmeuw/vivarium_research/pull/1041/files#diff-45a09ee257e7b14d6545111feb3c2f747ef32aa90038f0dbc028e55d2ff08f0dR421), where I think you are referring to the relationship attribute, not the tracked parent attribute (which of course is not set at initialization).

Reviewer 2: i agree with zeb --- we would like to avoid asking you guys to specifically delineate how the state table (or other within-sim machinery) is explicitly implemented, and instead have the docs outline the concepts you want to exist in the sim / maybe the needs they meet

i think sometimes those lines blur a bit, sorry about that!

i also agree disambiguation would be helpful here --- i think there are 3ish concepts that are getting conflated here:
1. a relationship within the sim we are calling "guardian". currently, the only usage i think you guys have for this is to decide who claims whom in tax returns (will it have other purposes?)
2. a relationship within the sim called "parent". currently, this is a link between the mother of a sim who was born during the simulation, and that child
3. a value in the "relationship to head of household" column called "parent"

it would be really helpful if, each time one of these was referenced, it was made explicit which one you are talking about (for example in lines 423-425)

Author: Okay I went through and tried to clarify this section and the language used. Let me know if there are still outstanding questions!

    * If there are not any other non-relatives in the house, make "N/A"
- Child is the reference person 
    * If someone has a defined [parent] relationship, assign them as guardian 
    * Otherwise, assign another relative (anyone who is NOT a roommate/housemate or other nonrelative in the same house) who is between 20 and 45 years older than the child. If there are multiple, assign at random.
    * If there are no other relatives in the house, make "N/A"

Once a guardian is assigned, if there is a spouse or unmarried partner 
for the guardian simulant (reference person and spouse/unmarried partner ONLY), then 
include both as guardians. Otherwise only include the one as a guardian. 

**For a simulant who is below 24 and in GQ at college:**

Simulant will be randomly assigned to a guardian based on the below rules: 

- 78.5% will be assigned to a guardian within their state. The remainder will be assigned out of state source1_. For early versions with only one state, the out of state guardians can be ignored. 
- Match to a person 20 to 45 years older than the child 

> - If child is not "Multiracial or Some Other Race", match parent's race. If child is "Multiracial or Some Other Race", then assign to a guardian of any race

Reviewer: ```suggestion
- If child is not "Multiracial or Some Other Race", match guardian's race. If child is "Multiracial or Some Other Race", then assign to a guardian of any race
```

Author: Good catch!

#. We have "parents" fall between 20-45 older than the child. This is an oversimplification. Some parents (especially men) fall outside of this range. Also some age gaps are more common than others. 
#. The only people who are seen as "in college" are in GQ in college. Plenty of people attend college from home, but we do not track education so are not accounting for this. 
#. We assign GQ college folks to "guardians" within an age limit. Some are likely supported by a grandparent or other person outside of our qualifications, but this is not included. 

Guardian(s) on Time Steps
^^^^^^^^^^^^^^^^^^^^^^^^^

For those who had a guardian initialized at the start of the 
simulation, that assignment will not change. If the guardian 
moves, they will remain as the guardian. If they die, the 
simulant will not have a living guardian. 

A person cannot "age out" of their guardian when they become 
an adult or have their own children. Therefore it is possible 
to both HAVE a guardian and BE a guardian. 

However, for simulants born in the simulation, they receive a 
tracked mother (term for the person that births them, regardless 
of gender). 

> The tracked mother will be assigned as a guardian. In addition, if 

Reviewer: I think it is good to have this here, but we might want to have it in the fertility section as well? Or reference it from that section? From an engineering perspective, this will be implemented in that component, I believe.

Author: Hmm good point, I'll add that in!

the "reference person" relationship field, we use the following
mapping:

> In addition, tracked mothers will be assigned as "guardians" based on the 

Reviewer: I think you'll want to move this above the previous paragraph, so it doesn't go between `the following mapping:` and the mapping.

Author: Thanks!

    * The reference person will submit the form, the spouse will be listed as the joint filer. 
    * There does not need to be persistence in who files jointly, it can be re-drawn each year. 
- All other non-married simulants in a household with a W2 or 1099 will file separately, based on the income rules above (e.g., a low-income earner in a house with other earners will be randomly assigned to file or not file, independent of others in the household). Please note that simulants can BOTH be claimed as a dependent AND file their own taxes. 
- All simulants eligible to be dependents will be claimed. To be eligible, the simulant must either: 

>     * Be under the age of 19 

Reviewer 3: Is this less than or equal to 18 or less than 18?

Author: Added clarification

    * OR be less than 24, in GQ in college, and earn less than $10,000 

>     * OR be any age, but NOT "housemate/roommate" or "other nonrelative" to whoever is claiming them and earn less than $4300 

Reviewer: There are two issues here, neither of which is new but which I am only noticing now:
- By using "whoever is claiming them" here, this part depends on the next, but the next depends on this part ("An **eligible** simulant...").
- "Whoever is claiming them" may live in a different household, in which case the "housemate/roommate" or "other nonrelative" part is not something we can check.

Happy to help think through how to clarify this rule, if you want.

- An eligible simulant will be claimed by: 
    * The defined "guardian" from initialization OR "parent" from birth (Note: if there are two parents/guardians and they are not filing jointly, assign the simulant randomly to one)

>     * If there is not a parent/guardian tracked, they will randomly be assigned to a tax-filing relative (not housemate or other non-relative) in the household 

Reviewer: Won't there be cases where this person is e.g. in college GQ (where we can't find relatives) and not have a tracked parent or guardian? For example if they started the sim in a different type of GQ, or were living in a household where we couldn't identify a guardian (i.e. the "N/A" case). I think we need to have a last option here, where if there are no tax-filing relatives, they are not claimed at all (though this contradicts the "All simulants eligible to be dependents will be claimed" statement above).

Reviewer: Also, as written, it seems like this line assumes that the simulant is a relative of the reference person. Or was your intention that even if they were e.g. "roommate" they would be claimed by a relative of the reference person?

Author: You're right, I added a line where if there is not anyone to claim them they remain unclaimed 

I made sure to clarify in line 1915 that to qualify as a dependent they had to be a relative of the reference person (so a roommate only wouldn't qualify to be claimed. This WILL miss people who might be related to someone in the house but just not the reference person, but I think this will be reasonably rare.

    * The reference person will submit the form, the spouse will be listed as the joint filer. 
    * There does not need to be persistence in who files jointly, it can be re-drawn each year. 
- All other non-married simulants in a household with a W2 or 1099 will file separately, based on the income rules above (e.g., a low-income earner in a house with other earners will be randomly assigned to file or not file, independent of others in the household). Please note that simulants can BOTH be claimed as a dependent AND file their own taxes. 
- Simulants eligible to be dependents must qualify as ONE of the following: 
    * Be under the age of 19 (less than or equal to 18)
    * OR be less than 24, in GQ in college, and earn less than $10,000 

>     * OR be any age, but be either a relative (NOT "housemate/roommate" or "other nonrelative") of the reference person, or be the reference person and earn less than $4300 

Reviewer: I think we can simplify this further (see suggestion below).
```suggestion
    * OR be any age, but earn less than $4300 
```

Author: Thanks! This does look simpler

- An eligible simulant will be claimed in the below order of options: 
    * The defined "guardian" (Note: if there are two parents/guardians and they are not filing jointly, assign the simulant randomly to one) 

>     * If there is not a "guardian" they will be randomly assigned to any relative (reference person OR relative of the reference person (NOT roommate or non-relative)) or a relative of themselves (if they are the reference person) 

Reviewer: ```suggestion
    * Otherwise, if they are the reference person or a relative of the reference person (NOT roommate or non-relative) they will be randomly assigned to someone else in the household who is the reference person or a relative of the reference person (NOT roommate or non-relative)
```

>     *  If there is no one found to claim them, they will remain unclaimed. This is most common for sims in GQ's but there might be other cases

Reviewer: ```suggestion
    *  If there is no one found to claim them, they will remain unclaimed. This is most common for sims in GQ's or non-relatives in households but there might be other cases
```
